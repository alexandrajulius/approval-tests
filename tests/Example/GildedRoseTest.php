<?php

declare(strict_types=1);

namespace AHJ\ApprovalTests\Tests\Example;

use AHJ\ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;

final class GildedRoseTest extends TestCase
{
    public function testUpdateQuality(): void
    {
        $input = [
            new Item('foo', 0, 1),
            new Item('bar', 1, 1),
            new Item('zero', 1, 0),
            new Item('zorg', 1, 100),
        ];

        $actual = (new GildedRose())->updateQuality($input);

        Approvals::create()->verifyList($input, $actual);
    }
}
