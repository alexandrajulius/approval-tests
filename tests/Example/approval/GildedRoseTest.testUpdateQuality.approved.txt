[foo, 0, 1] -> [foo, -1, 0]
[bar, 1, 1] -> [bar, 0, 0]